bot/resources/tags/type-hint.md

```diff
@@ -0,0 +1,19 @@
+**Type Hints**
+
+A type hint indicates what type something is expected to be. For example,
+```python
+def add(a: int, b: int) -> int:
+    return a + b
+```
+In this case, we have a function,`add`, with parameters `a` and `b`. The type hints indicate that the parameters and return type are all integers.
+
+It's important to note these are just hints and have no runtime effect. For example,
+```python
+# Uh oh
+add("hello ", "world")
+```
+This code won't error even though it doesn't follow the function's type hints. It will just concatenate the two strings.
+
+Third party tools like [mypy](http://mypy-lang.org/) can enforce your type hints. Mypy would error in the second example.
+
+For more info about type hints, check out [PEP 484](https://www.python.org/dev/peps/pep-0484/).
```

Comment:
Not sure if this would be out of scope for this tag but I think it'd be a nice addition to reference https://typing.readthedocs.io/en/latest/index.html as a more general user focused resource on static typing as it'd be next logical step after learning about type-hints IMO.

Reply:
Yes, there are plans for adding more content like that but it's limited by people's time.

Reply:
Included it, I also mentioned the `mypy` docs as I think that's probably one of the best places to go for an introduction to type hinting.